jenkins/debian/debos/scripts/crush.sh

```diff
@@ -6,151 +6,181 @@ set -e
-"insserv "\
-"init-system-helpers "\
-"cpio "\
-"passwd "\
-"libsemanage1 libsemanage-common "\
-"libsepol1 "\
-"gzip "\
-"gnupg "\
-"gpgv "\
-"hostname "\
-"adduser "\
-"debian-archive-keyring "\
 
+filesystem()  {
+   UNNEEDED_PACKAGES="libfdisk1 cpio hostname gzip "
 
-# Removing unneeded packages
-for PACKAGE in ${UNNEEDED_PACKAGES}
-do
-	echo "Forcing removal of ${PACKAGE}"
-	if ! dpkg --purge --force-remove-essential --force-depends "${PACKAGE}"
-	then
-		echo "WARNING: ${PACKAGE} isn't installed"
-	fi
-done
+   remove_packages "${UNNEEDED_PACKAGES}"
+
+   # Partition and file system tools
+   rm -f usr/sbin/*fdisk
+   rm -f usr/sbin/mkfs*
+   rm -f usr/sbin/fsck*
+
+   # No need for fuse
+   find usr etc -name '*fuse*' -prune -exec rm -r {} \;
+
+   rm -f usr/bin/hostnamectl
+}
+
+package_management()  {
+   UNNEEDED_PACKAGES="apt libapt-pkg5.0 debconf libdebconfclient0 debian-archive-keyring "
+
+   remove_packages "${UNNEEDED_PACKAGES}"
+
+   # Show what's left package-wise before dropping dpkg itself
+   COLUMNS=300 dpkg -l
 
-# Show what's left package-wise before dropping dpkg itself
-COLUMNS=300 dpkg -l
+   # Drop dpkg
+   dpkg --purge --force-remove-essential --force-depends  dpkg
 
-# Drop dpkg
-dpkg --purge --force-remove-essential --force-depends  dpkg
+   # No apt or dpkg, no need for its configuration archives
+   rm -rf etc/apt
+   rm -rf etc/dpkg
+}
 
-# No apt or dpkg, no need for its configuration archives
-rm -rf etc/apt
-rm -rf etc/dpkg
+systemd()  {
+   # Unused systemd generators
+   rm -f lib/systemd/system-generators/systemd-cryptsetup-generator
+   rm -f lib/systemd/system-generators/systemd-debug-generator
+   rm -f lib/systemd/system-generators/systemd-gpt-auto-generator
+   rm -f lib/systemd/system-generators/systemd-hibernate-resume-generator
+   rm -f lib/systemd/system-generators/systemd-rc-local-generator
+   rm -f lib/systemd/system-generators/systemd-system-update-generator
+   rm -f lib/systemd/system-generators/systemd-sysv-generator
 
-# Drop directories not part of ostree
-# Note that /var needs to exist as ostree bind mounts the deployment /var over
-# it
-rm -rf var/* srv share
+   # Efi blobs
+   rm -rf usr/lib/systemd/boot
 
-# ca-certificates are in /etc drop the source
-rm -rf usr/share/ca-certificates
+   # Translation catalogs
+   rm -rf usr/lib/systemd/catalog
 
-# No bash, no need for completions
-rm -rf usr/share/bash-completion
+   # Misc systemd utils
+   rm -f usr/bin/bootctl
+   rm -f usr/bin/busctl
+   rm -f usr/bin/localectl
+   rm -f usr/bin/systemd-cat
+   rm -f usr/bin/systemd-cgls
+   rm -f usr/bin/systemd-cgtop
+   rm -f usr/bin/systemd-delta
+   rm -f usr/bin/systemd-detect-virt
+   rm -f usr/bin/systemd-mount
+   rm -f usr/bin/systemd-path
+   rm -f usr/bin/systemd-run
+   rm -f usr/bin/systemd-socket-activate
 
-# No zsh, no need for comletions
-rm -rf usr/share/zsh/vendor-completions
+   # Systemd dns resolver
+   #find usr etc -name '*systemd-resolve*' -prune -exec rm -r {} \;
```

Comment:
Rather than keeping this as a comment, maybe it should be in a separate function that we can decide to call if these files should be removed.

Reply:
moved them into separate function `systemd_dns_nw_ntp_tools`